Sources/Security/Signing.swift

```diff
@@ -259,7 +240,50 @@ extension Signing.SignatureParameters {
 
 private final class BundleToken: NSObject {}
 
-// MARK: - Data extensions
+private extension Signing {
+
+    @available(iOS 13.0, macOS 10.15, watchOS 6.0, tvOS 13.0, *)
+    static func createPublicKey(with data: Data) throws -> PublicKey {
+        return try Algorithm(rawRepresentation: data)
+    }
+
+    static func extractAndVerifyIntermediateKey(
+        from signature: Data,
+        publicKey: Signing.PublicKey
+    ) -> Signing.PublicKey? {
+        guard #available(iOS 13.0, macOS 10.15, watchOS 6.0, tvOS 13.0, *) else { return nil }
+
+        let intermediatePublicKey = signature.component(.intermediatePublicKey)
+        let intermediateKeyExpiration = signature.component(.intermediateKeyExpiration)
+        let intermediateKeySignature = signature.component(.intermediateKeySignature)
+
+        guard publicKey.isValidSignature(intermediateKeySignature,
+                                         for: intermediateKeyExpiration + intermediatePublicKey) else {
+            Logger.warn(Strings.signing.intermediate_key_failed_verification(signature: intermediateKeySignature))
+            return nil
+        }
+
+        let expirationDate = Date(daysSince1970: UInt32(littleEndian32Bits: intermediateKeyExpiration))
+
+        guard expirationDate.timeIntervalSince(Date()) >= 0 else {
+            Logger.warn(Strings.signing.intermediate_key_expired(expirationDate, intermediateKeyExpiration))
+            return nil
+        }
+
+        Logger.verbose(Strings.signing.intermediate_key_creating(expiration: expirationDate,
+                                                                 data: intermediatePublicKey))
+
+        do {
+            return try Self.createPublicKey(with: intermediatePublicKey)
```

Comment:
We considered caching this, but I just ran a benchmark on my device and the whole process (creating this key and verifying the original signature) takes 0.17 seconds.

Reply:
Sorry I meant 0.17ms!